src/constructs/cloudwatch/lambda-alarms.ts

```diff
@@ -0,0 +1,44 @@
+import { ComparisonOperator, MathExpression } from "@aws-cdk/aws-cloudwatch";
+import type { GuStack } from "../core";
+import type { GuLambdaFunction } from "../lambda";
+import type { GuAlarmProps } from "./alarm";
+import { GuAlarm } from "./alarm";
+
+export type LambdaMonitoring = NoMonitoring | ErrorPercentageMonitoring;
+
+export interface NoMonitoring {
+  noMonitoring: true;
+}
+
+export interface ErrorPercentageMonitoring
+  extends Omit<GuAlarmProps, "metric" | "threshold" | "comparisonOperator" | "evaluationPeriods"> {
+  toleratedErrorPercentage: number;
+  numberOfFiveMinutePeriodsToEvaluate?: number;
+  noMonitoring?: false;
+}
+
+interface GuLambdaAlarmProps extends ErrorPercentageMonitoring {
+  lambda: GuLambdaFunction;
+}
```

Comment:
Maybe one for a future PR but I wonder if the `noMonitoring` stuff is something that should live at a higher level as it's unlikely to be specific to Lambda monitoring and also it feels a bit odd to define a prop here which then isn't used. 

Maybe `NoMonitoring` should be exported from somewhere else and the `monitoringConfiguration` prop becomes `monitoringConfiguration: LambdaMonitoring | NoMonitoring`? Then I guess `ErrorPercentageMonitoring` could extend `GuMonitoringProps` to add the `noMonitoring?: false`.

Reply:
Agreed - this part of the code needs a little more thought as we cover more use-cases. I think I will go ahead and merge this as is for now, but perhaps we can refactor this slightly as part of adding additional lambda patterns.